### ISSUE TITLE
POSTing single observation is inserted as list

### ISSUE THREAD

**Issue by ridoo** [52North/connected-systems-pygeoapi#4]
POSTing an observation of type `Quantity` will be inserted as a list of observations:

https://github.com/directedproject-eu/pygeoapi_csa/blob/main-directed/connected-systems-api/provider/part2/part2.py#L178

This is because `_create_observation(..)` requires a `List[Observation]`:

https://github.com/52North/connected-systems-pygeoapi/blob/47fa63960d3f93532480ba22344a73fb5aa2d9cc/connected-systems-api/provider/part2/part2.py#L177

The result of a single observation would be

```json
{
    "id": "5d657427-5104-4bce-b956-4b6734f02fea",
    "datastream@id": "6dc77e0b-0cff-3215-8224-a316b392faa4",
    "resultTime": "2024-12-20T18:01:00+00:00",
    "result": [22.2000007629395]
}
```

Note the `"result": [22.2000007629395]` part instead of `"result": 22.2000007629395`.

**Comment by ridoo** [comment]
Perhaps, this is cause more from this location here (but did not dig deeper atm):

https://github.com/directedproject-eu/pygeoapi_csa/blob/4a656b5027966034fa9bf8e5dff3911bf4650262/connected-systems-api/provider/part2/part2.py#L353-L356

**Comment by SpeckiJ** [comment]
Yes this is probably an artifact from the "insert observation (batch)" request that is no longer present in the specification - a lot of the code can be simplified and only support insertion of a single observation at a time.

**Comment by hetanthakkar** [comment]
Fixed in commit 850da2d ("Round and unpack single vs multiple values")

**Root cause:** `struct.unpack()` returns a tuple, which was serialized as an array in JSON.

**Fix:** Unpack the tuple and return a single value for Quantity types:
```python
first = round(next(iterator, None), 3)
rest = [round(v, 3) for v in iterator]
"result": first if not rest else [first] + rest
